### PR TITLE
Add basic Kirby Dream Land support

### PR DESCRIPTION
**Pull request by MaciekBaron**
Thought I'll add basic support for Kirby Dream Land:

- Ability to start game (skips start screen and intro)
- Tracking health, lives and score

### CODE REVIEW

**Comment by Baekalfen** [comment]
That's really cool! Awesome work!

Do you have a small example, that I could include in the automatic testing?

**Comment by MaciekBaron** [comment]
Yeah sure, I'll create an example in the `examples` folder, will that do?

**Comment by Baekalfen** [comment]
Yes, that would be perfect

**Comment by MaciekBaron** [comment]
Ok that's added one, I tested it locally and all assertions pass

File: pyboy/plugins/game_wrapper_kirby_dream_land.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        self._tile_cache_invalid = True
+        self._sprite_cache_invalid = True
+
+        multiplier = 100000
+        index = 0
+        score = 0
+        while multiplier > 10:
+          score += self.pyboy.get_memory_value(0xD06F + index) * multiplier
+          multiplier /= 10
+          index += 1
```
Comment (Baekalfen) [inline pyboy/plugins/game_wrapper_kirby_dream_land.py:64]: I can't tell what this part is doing

Comment (MaciekBaron) [reply pyboy/plugins/game_wrapper_kirby_dream_land.py:64]: The score is stored between `0xd06F`and five bytes later, with every digit of the score taking one byte (later used for displaying the score on the screen).

So if the score is `007250`, I multiply each digit (`0`, `0`, `7`, `2` etc.) by the multiplier (`100000`, `10000`, `1000`, `100` etc.) to get the score value.

Hope this makes sense - there might be a better way of achieving this - I think you already employ a similar approach when reading the score from the screen and using the tile indexes.

Comment (Baekalfen) [reply pyboy/plugins/game_wrapper_kirby_dream_land.py:64]: Ah, of course! At a glance, I couldn't figure out what was going to be exponential like that.

**Comment by Baekalfen** [comment]
I fixed a few minor things. I'm no AI/ML expert, but is it important to have the level progressing (how far to the right the AI gets) as a component of the `fitness` value?

If not, I think I'm ready to merge.

And just a question. How did you locate those memory addresses?

**Comment by MaciekBaron** [comment]
The fitness value is currently just a "placeholder", can be improved later on.
I got the memory values using [BGB](https://bgb.bircd.org/) (an emulator with memory viewing/editing capabilities).

**Comment by Baekalfen** [comment]
Ok, understood. But thanks a lot for your contribution! It's pretty solid work :) 

I'll merge it into a pre-release branch now and hopefully soon deploy it as v1.0.2.

**Pull request merged**